clouddriver-lambda/src/main/java/com/netflix/spinnaker/clouddriver/lambda/provider/agent/AmazonLambdaFunctionCachingAgent.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.lambda.provider.agent;
+
+import com.amazonaws.services.lambda.AWSLambda;
+import com.amazonaws.services.lambda.model.*;
+import com.netflix.spinnaker.cats.agent.*;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.cache.DefaultCacheData;
+import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.aws.provider.AwsInfrastructureProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.lambda.cache.Keys;
+import groovy.util.logging.Slf4j;
+import org.codehaus.groovy.runtime.DefaultGroovyMethods;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.amazonaws.services.lambda.model.ListEventSourceMappingsResult;
+import com.amazonaws.services.lambda.model.ListEventSourceMappingsRequest;
+
+import java.util.*;
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
+import static com.netflix.spinnaker.clouddriver.lambda.cache.Keys.Namespace.LAMBDA_NAME;
+import static com.netflix.spinnaker.clouddriver.lambda.cache.Keys.Namespace.IAM_ROLE;
+
+@Slf4j
+public class AmazonLambdaFunctionCachingAgent implements CachingAgent, AccountAware {
+
+  private final AmazonClientProvider amazonClientProvider;
+  private final NetflixAmazonCredentials account;
+  private final String region;
+  private static final Set<AgentDataType> types = Collections.unmodifiableSet(DefaultGroovyMethods.asType(new ArrayList<AgentDataType>(Arrays.asList(AUTHORITATIVE.forType(LAMBDA_NAME.ns))), Set.class));
```

Comment:
Would this work instead?

    private static final Set<AgentDataType> types = Collections.singleton(AUTHORITATIVE.forType(LAMBDA_NAME.ns));